### DISCUSSION TITLE
Potentionally over crowding upload folder

### DISCUSSION THREAD

**Discussion by stormbyte** [directus/directus#9701] [Archive]
In my current I have uploaded 38 images.
Every image has between 4 and 6 resized versions resulting in 254 files in the upload folder.

If this behavior continues then this could result in slow performance serving these files.
Seperating these files in folders will keep these files fast.
The file:
`fca13e93-79a4-4fe0-9b15-4eca07f65131.jpeg`
Could be written in a structure like:
`/f/c/fca13e93-79a4-4fe0-9b15-4eca07f65131.jpeg`

### To Reproduce

Upload a image into Directus.
All files will be written in uploads folder.

**Comment by rijkvanzanten** [comment]
_Moving this to a feature request discussion, as everything is technically working as expected/intended._

For folders, I think the main question we're solving for right now is what sort of configuration to open up on the "physical" folders side of things (next to Directus' virtual folders), see for example https://github.com/directus/directus/discussions/8950.

With regards to the maximum number of files in a folder, are you running into any issues currently around the number of files? Unless you're working with hundreds of thousands of files, the number of files shouldn't pose an issue 🤔

**Reply by stormbyte** [reply]
I agree that directus is currently working as intended.
The solution In #8950 will solve a potentional problem.

Guess having 2 threads pointing to the same issue is pointless

**Reply by rijkvanzanten** [reply]
Yeah lets keep discussing in #8950 👍🏻
